state-chain/custom-rpc/src/lib.rs

```diff
@@ -280,6 +280,22 @@ pub struct RpcEnvironment {
 	pools: PoolsEnvironment,
 }
 
+#[derive(Serialize, Deserialize, Clone, PartialEq, Eq, Debug)]
+pub struct PoolPriceV2 {
+	pub from_asset: RpcAsset,
+	pub to_asset: RpcAsset,
+	pub price: PoolPrice,
```

Comment:
I'd suggest, flatten the PoolPrice into this struct (I dont mean serde(flatten)).

Reply:
Also adding the "V1" to the old types would be nice.